argus/backend/service/views.py

```diff
@@ -0,0 +1,232 @@
+import datetime
+import logging
+import re
+from functools import partial, reduce
+from typing import Any, Callable, TypedDict
+from urllib.parse import unquote
+from uuid import UUID
+
+from cassandra.cqlengine.models import Model
+from argus.backend.models.web import ArgusGroup, ArgusRelease, ArgusTest, ArgusUserView, User
+from argus.backend.plugins.loader import all_plugin_models
+from argus.backend.util.common import chunk, current_user
+
+LOGGER = logging.getLogger(__name__)
+
+
+class UserViewException(Exception):
+    pass
+
+
+class ViewUpdateRequest(TypedDict):
+    name: str
+    description: str
+    display_name: str
+    tests: list[str]
+    widget_settings: str
+
+
+class UserViewService:
+    ADD_ALL_ID = UUID("db6f33b2-660b-4639-ba7f-79725ef96616")
+    def create_view(self, name: str, items: list[str], widget_settings: str, description: str = None, display_name: str = None) -> ArgusUserView:
+        view = ArgusUserView()
+        view.name = name
```

Comment:
what happens if someone posts the same name? Won't it override?

Reply:
Hmm, currently the sluggified name is not really used - it won't override as the partition key is the `id` column. To be honest I'm not sure we really need the view name, maybe I could output as a helper for the view path e.g. `/views/<uuid>/<slug>` just to make it easier for user

Reply:
uuid's in path are not helping and are generally looking bad.
can't we enforce slug (name) to be unique and make /view/<slug> path?

Reply:
We can, will do that.

Reply:
gh, cut `<slug>` from my comment above (fixed), I hope you got it

Reply:
Looks great now.

😅 polish symbols match, how about Cyrillic?

Reply:
Non latin scripts just get stripped unfortunately, I think the library supports custom matching dictionaries but it's its default behaviour.

![image](https://github.com/scylladb/argus/assets/7761415/7dade81b-5777-4456-a6d7-f41d0c8b9882)

Reply:
no problem, anyway we don't expect using different language than english there